### ISSUE TITLE
Fowarding to Logstash stop working after a configuration update

### ISSUE THREAD

**Issue by tchlyah** [jhipster/generator-jhipster#7095]
##### **Overview of the issue**

After any configuration update on Consul or jHipster Registry, the Logstash appender stop working. the neither `@RefreshScope` (#6679) annotation nor `LoggerContextListener` (#4334) solve the issue.

##### **Reproduce the error**

1. Start jhipster Console
1. Start a microservice with either Consul or jhipster Registry
1. You can see microservice logs in Kibana
1. Change configuration
1. Microservice stop forwarding to Logstash

##### **Related issues**

#6668 
#6402

##### **JHipster Version(s)**

4.13.3

##### **JHipster configuration**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "web.common",
      "nativeLanguage": "fr"
    },
    "jhipsterVersion": "4.13.3",
    "baseName": "microservice",
    "packageName": "web.common",
    "packageFolder": "web/common",
    "serverPort": "8084",
    "authenticationType": "uaa",
    "uaaBaseName": "uaa",
    "hibernateCache": "no",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "no",
    "devDatabaseType": "no",
    "prodDatabaseType": "no",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": "consul",
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "enableTranslation": true,
    "applicationType": "microservice",
    "testFrameworks": [
      "gatling",
      "cucumber"
    ],
    "jhiPrefix": "jhi",
    "skipClient": true,
    "skipUserManagement": true,
    "nativeLanguage": "fr",
    "languages": [
      "fr",
      "en"
    ],
    "clientPackageManager": "npm",
    "cacheProvider": "no",
    "enableHibernateCache": false
  }
}
</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b15, mixed mode)
Picked up _JAVA_OPTIONS: -Djava.net.preferIPv4Stack=true

git version 2.14.1.windows.1

node: v6.10.2

npm: 4.5.0

yeoman: 2.0.0

Docker version 17.06.2-ce, build cec0b72

docker-compose version 1.14.0, build c7bdf9e3
```

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by PierreBesson** [comment]
This is really too bad as we already spent quite some time trying to fix this issue.
Maybe the problem is only for Consul and due to the @conditionalonconsul annotation.
@tchlyah Does it work if you remove this conditional annotation ?

Can anyone try it with Eureka ?

**Comment by tchlyah** [comment]
Yes I can confim that it is also not working with Eureka, the only difference is that with it, the configuration refresh is not automatic, you should do a POST request to `/management/refresh` to trigger it.

Using `@RefreshScope` with `@ConditionalOnConsul` annotation causes the exception only if Consul is disabled, it has nothing to do with loosing logs.

**Comment by tchlyah** [comment]
Any one can help? We have serious production issues because of that, we even loose our logs if there is a small disconnection from Consul!

**Comment by PierreBesson** [comment]
@tchlyah I suggest that you remove the @RefreshScope annotation. I think you encounter problems as Spring cloud consul auto-reload the application context when it detect config changes and it seems that small downtimes are causing the app to receive the event for you.
Removing this annotation is not a big deal for me you will just need to restart the app to take configuration changes into account.
Otherwise there might be a specific property under spring.cloud.consul.config that let's you disable the refresh behavior.

**Comment by tchlyah** [comment]
I've tested both with and without @RefreshScope, it doesn't change anything!
I'm not aware of such property, it will be helpful if it exists

**Comment by PierreBesson** [comment]
What you are looking for is documented here : https://github.com/spring-cloud/spring-cloud-consul/blob/master/docs/src/main/asciidoc/spring-cloud-consul.adoc#config-watch
Try setting this: `spring.cloud.consul.config.watch.enabled=false`.
If this fixes your problem, pease do a PR.

I think this is a newer feature of spring cloud consul (enabled by default !) so this might explain why we didn't have the issue before.

**Comment by PierreBesson** [comment]
@tchlyah Any update to your problem ?
